### ISSUE TITLE
Don't Use Color Scheme for Block Selection

### ISSUE THREAD

**Issue by shaunandrews** [WordPress/gutenberg#27411]
When selecting a block in the editor an outline often appears to indicate the shape of the block. The color of this outline matches the keyboard focus state of UI within the editor. Here's what the outline looks like across all the Core color schemes:

<img width="933" alt="image" src="https://user-images.githubusercontent.com/191598/100792113-27e9e000-33e8-11eb-95eb-560632198e18.png">

As we consider using color to indicate special blocks, like a Template Part, we run into issues around which color we can use. On top of this, the theme's so-called "spot" color is used throughout the UI to indicate primary buttons, keyboard focus, and other indicators of interactivity. By also using this color to indicate block selection, we muddy the understanding around the use of color.

In addition, using color _within the canvas_ can lead to clashes with a theme or blocks defined colors. For example, if a block sits on an orange background and the color scheme defines orange as the "spot" color then the block selection outline becomes nearly invisible:

<img width="534" alt="image" src="https://user-images.githubusercontent.com/191598/100792794-3258a980-33e9-11eb-9c03-7e08435c6987.png">

I think it could make sense to stop relying on the user's select color scheme to indicate selected blocks. It might be better to use a standard color for on-canvas UI highlighting, and fine tune this color to best avoid clashing with theme and block-defined colors.

Here's what the outline could look like if it used purple, yellow (which Google seems to indicate are not very popular colors), and black. I've also added an additional white outline to help separate the indicator from any background colors:

<img width="440" alt="image" src="https://user-images.githubusercontent.com/191598/100793890-d000a880-33ea-11eb-9fb4-76ac2d778f2e.png">

--

I know there is some intentional overlap between the block outline and the keyboard focus, but I'm not sure if that's actually helping or hurting things; I often hear about confusion related to the keyboard focus state and wonder if its due to the similarities with block selection.
